UpsertUserDataProcessing/handler.ts

```diff
@@ -91,52 +94,69 @@ export function UpsertUserDataProcessingHandler(
           : UserDataProcessingStatusEnum.PENDING;
       }
     );
-    const userDataProcessing = UserDataProcessing.decode({
+    const userDataProcessing: UserDataProcessing = {
       choice: upsertUserDataProcessingPayload.choice,
       createdAt: new Date(),
       fiscalCode,
       status: computedStatus,
       userDataProcessingId: id
-    });
+    };
+    const errorOrUpsertedUserDataProcessing = await userDataProcessingModel.createOrUpdateByNewOne(
+      userDataProcessing
+    );
+    if (isLeft(errorOrUpsertedUserDataProcessing)) {
+      const { body } = errorOrUpsertedUserDataProcessing.value;
 
-    if (isLeft(userDataProcessing)) {
-      const error = userDataProcessing.value;
-      return ResponseErrorFromValidationErrors(UserDataProcessing)(error);
-    } else {
-      const errorOrUpsertedUserDataProcessing = await userDataProcessingModel.createOrUpdateByNewOne(
-        userDataProcessing.value
-      );
+      context.log.error(`${logPrefix}|ERROR=${body}`);
 
-      context.log.error(
-        `errorOrUpsertedUserDataProcessing ${errorOrUpsertedUserDataProcessing}`
+      return ResponseErrorQuery(
+        "Error while creating a new user data processing",
+        errorOrUpsertedUserDataProcessing.value
       );
-      if (isLeft(errorOrUpsertedUserDataProcessing)) {
-        const { body } = errorOrUpsertedUserDataProcessing.value;
-
-        context.log.error(`${logPrefix}|ERROR=${body}`);
+    }
 
-        return ResponseErrorQuery(
-          "Error while creating a new user data processing",
-          errorOrUpsertedUserDataProcessing.value
+    const createdOrUpdatedUserDataProcessing =
+      errorOrUpsertedUserDataProcessing.value;
+    const errorOrMaybeRetrievedProfile = await profileModel.findOneProfileByFiscalCode(
```

Comment:
Since this is a web controller it's better to mode `findOneProfileByFiscalCode` into the activity to leave the controller responsive / fast. is it possible ?